cfme/infrastructure/provider/rhevm.py

```diff
@@ -57,27 +39,128 @@ def deployment_helper(self, deploy_args):
 
     @classmethod
     def from_config(cls, prov_config, prov_key, appliance=None):
-        credentials_key = prov_config['credentials']
-        credentials = {
-            # The default credentials for controlling the provider
-            'default': cls.process_credential_yaml_key(credentials_key),
-        }
+        endpoints = [RHEVMEndpoint(**prov_config['endpoints']['default']),
+                     CANDUEndpoint(**prov_config['endpoints']['candu'])]
         if prov_config.get('discovery_range', None):
             start_ip = prov_config['discovery_range']['start']
             end_ip = prov_config['discovery_range']['end']
         else:
             start_ip = end_ip = prov_config.get('ipaddress')
-        if prov_config.get('candu_credentials', None):
-            # Insert C&U credentials if those are present
-            credentials['candu'] = RHEVMProvider.process_credential_yaml_key(
-                prov_config['candu_credentials'], cred_type='candu')
+
         return cls(name=prov_config['name'],
-                   hostname=prov_config['hostname'],
-                   ip_address=prov_config['ipaddress'],
-                   api_port='',
-                   credentials=credentials,
+                   endpoints=endpoints,
                    zone=prov_config.get('server_zone', 'default'),
                    key=prov_key,
                    start_ip=start_ip,
                    end_ip=end_ip,
                    appliance=appliance)
+
+    def create(self, cancel=False, validate_credentials=True, check_existing=False,
```

Comment:
**QUESTION:** Why multiple create methods here, are they not all pretty identical, I would really like to avoid this if we can. The methods _look_ similar, but would like a quick answer here as to why we can't use a common one. That just fills the main form an then tries to loop fill the credentials.